IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs

```diff
@@ -72,18 +73,47 @@ public async Task<IReadOnlyCollection<IWebElement>> FindElements(By by, int seco
             {
                 return await Wait.Until<NoSuchElementException, StaleElementReferenceException, IReadOnlyCollection<IWebElement>>(() => Driver.FindElements(by), TimeSpan.FromSeconds(secondsToWait));
             }
-            catch(AggregateException)
+            catch (AggregateException)
             {
                 return Array.Empty<IWebElement>();
             }
         }
 
+        public async Task<bool> IsElementInExpectedState(Func<bool> func, bool expectedResult = true, int secondsToWait = 15)
```

Comment:
This was primarily to get around knowing when to return right away with either a true or false scenario, and how to handle a throw. The two specific considerations:
1) Regardless of if the Selenium property is true or false, how do we know when to continue to re-evaluate the delegate? If I call this expecting a true scenario, but the Selenium property is false, do we immediately return false and potentially fail a test, or continue to retry until the timeout is reached, and how do we know when to handle the inverse case.
2) When specific exceptions are thrown, how do we know if they satisfy the true or false scenario (e.g. NoSuchElementException is functionality equivalent to !element.Displayed for the purposes of checking if an element isn't present on screen. But depending on how the caller rights their delegate, how do we know to return true or false? Which, actually, I forgot to add that check into this version of code. Oops!)

OR, to perhaps phrase this a different way... is it safe to make the assumption that the caller will know not to invert the delegate response itself, as that would cause an exception or false scenario to need to be returned "true"? So E.G. these three scenarios are supported:
Assert.True(IsElementInExpectedState(() => element.Displayed)); // both hidden and exception cases return false
Assert.False(IsElementInExpectedState(() => element.Displayed)); // same as above
Assert.True(!IsElementInExpectedState(() => element.Displayed)); // same as above but inverted

But the following isn't:

Assert.True(IsElementInExpectedState(() => !element.Displayed)); // The exception case will return false even though it's functionality equivalent to what we're asking for to be true

Reply:
Oh, and an addendum: in the first two assert scenarios above, the function will only ever wait for the true scenario even though the false is the one we expect in the second. That was the other trouble we had on the PTT without passing in an expected result.

Would it make more sense for two separate methods? Something like: IsElementInValidState (existing, visible, and interactable, which is the primary use case for the method,) and IsElementInInvalidState (basically... not any of the above.)